lib/galaxy/tool_util/xsd/galaxy.xsd

```diff
@@ -4314,6 +4314,25 @@ template if the parameter is ``false`` or not checked by the user. Used only
 when the ``type`` attribute value is ``boolean``.</xs:documentation>
           </xs:annotation>
         </xs:attribute>
+        <xs:attribute name="allow_uri_if_prefixed" type="xs:string">
+          <xs:annotation>
+            <xs:documentation xml:lang="en"><![CDATA[
+When using `deferred` datasets, Galaxy will try to materialize them into files when
+running the tool. In case we don't want to download them. The `allow_uri_if_prefixed` attribute
+can be used to avoid materialization and pass the URI as is to the tool.
+This is useful when the tool can handle the URI directly. Example:
+
+```xml
+<param name="input" type="data" format="txt" allow_uri_if_prefixed="http|s3" />
```

Comment:
Maybe we should be more specific/strict, and only allow exact protocols to be specified?

```suggestion
<param name="input" type="data" format="txt" allow_uri_if_protocol="http|https|s3" />
```

Reply:
Sounds good to me :+1: Thanks!

Reply:
Is there a reason to use | instead of a comma? For startups we also use commas

Reply:
> Is there a reason to use | instead of a comma? For startups we also use commas

No reason beyond following the ideas in https://github.com/galaxyproject/galaxy/issues/19065, of course, we can use commas for consistency although I kind of like the `|` as lately there is more tendency to use it to describe unions like in [PEP604](https://peps.python.org/pep-0604/) or [Typescript](https://www.typescriptlang.org/docs/handbook/typescript-in-5-minutes-func.html#unions). 

That said, I'll change it to commas for consistency with the existing way :+1: